airbyte-integrations/connectors/source-mixpanel/source_mixpanel/source.py

```diff
@@ -115,24 +116,40 @@ def streams(self, config: Mapping[str, Any]) -> List[Stream]:
         logger.info(f"Using start_date: {config['start_date']}, end_date: {config['end_date']}")
 
         auth = self.get_authenticator(config)
-        streams = []
-        for stream in [
-            Annotations(authenticator=auth, **config),
-            Cohorts(authenticator=auth, **config),
-            Funnels(authenticator=auth, **config),
-            Revenue(authenticator=auth, **config),
-            CohortMembers(authenticator=auth, **config),
-            Engage(authenticator=auth, **config),
+        streams = [
             Export(authenticator=auth, **config),
-        ]:
+        ]
+
+        # we only make calls to one stream of each API type to save time, based on the assumption that
+        # if one stream of the API type is available, all others are available as well and vice versa
+
+        streams_by_api_types = [
+            [
+                Annotations,
+            ],
+            [Engage, Cohorts, CohortMembers, Funnels, Revenue],
+        ]
+        stream_kwargs = {"authenticator": auth, **config}
+        if not os.environ.get("SLEEP_BETWEEN_CHECK_AND_DISCOVER_REQUESTS", False):
```

Comment:
nit: For simplicity we may want `SLEEP_BETWEEN_CHECK_AND_DISCOVER_REQUESTS` to be more directly related to `reqs_per_hour_limit`

For example if it was `REQS_PER_HOUR_LIMIT` and the default value was 0 then during testing we just set

```py
custom_environment_variables:
  REQS_PER_HOUR_LIMIT: 60
  # ...
```

Reply:
done